packages/core/strapi/lib/services/entity-service/components.js

```diff
@@ -77,18 +80,18 @@ const createComponents = async (uid, data) => {
         throw new Error('Expected an array to create repeatable component');
       }
 
-      componentBody[attributeName] = await Promise.all(
-        dynamiczoneValues.map(async (value) => {
-          const { id } = await createComponent(value.__component, value);
-          return {
-            id,
-            __component: value.__component,
-            __pivot: {
-              field: attributeName,
-            },
-          };
-        })
-      );
+      // No Promise.all to avoid deadlocks
+      componentBody[attributeName] = [];
+      for (const value of dynamiczoneValues) {
+        const { id } = await createComponent(value.__component, value);
+        componentBody[attributeName].push({
+          id,
+          __component: value.__component,
+          __pivot: {
+            field: attributeName,
+          },
+        });
+      }
```

Comment:
We don't do that in the code base for the moment but maybe we could. What do you think of bringing the subject to the backend sync?

Reply:
What subject ? Handle sequential promises with a function instead of using `for(const ... in ...) {}` ? Don't know how to name this subject 😅

Reply:
I more used to the for loop on my side so I would not change it. However I'm opened to do something else if we want to make the code style evolve in that direction

Reply:
For instance it looks like an `async reduce`, we can have an `async map` and an `async forEach`.
So I see this discussion a bit bigger than just this piece of code in this PR. If we do a `utils`, since it's to be used in the rest of the code base, I would like to discuss it with the rest of the team